src/pyuvsim/simsetup.py

```diff
@@ -1911,129 +2014,108 @@ def initialize_uvdata_from_params(
     if "Npols" not in uvparam_dict:
         uvparam_dict["Npols"] = len(uvparam_dict["polarization_array"])
 
-    uvparam_name = "cat_name"
+    # telescope frame is set from world in parse_telescope_params.
+    # Can only be itrs or (if lunarsky is installed) mcmf
+    if tele_params["telescope_frame"] == "itrs":
+        telescope_location = EarthLocation.from_geocentric(
+            *tele_params["telescope_location"], unit="m"
+        )
+    elif tele_params["telescope_frame"] == "mcmf":
+        telescope_location = MoonLocation.from_selenocentric(
+            *tele_params["telescope_location"], unit="m"
+        )
+        telescope_location.ellipsoid = tele_params["ellipsoid"]
 
     if "cat_name" not in param_dict and "object_name" not in param_dict:
         tloc = EarthLocation.from_geocentric(
-            *uvparam_dict["telescope_location"], unit="m"
+            *tele_params["telescope_location"], unit="m"
         )
-        time = Time(uvparam_dict["time_array"][0], scale="utc", format="jd")
+        time = Time(parsed_time_dict["time_array"][0], scale="utc", format="jd")
         src, _ = create_mock_catalog(time, arrangement="zenith", array_location=tloc)
         if "sources" in param_dict:
             source_file_name = os.path.basename(param_dict["sources"]["catalog"])
-            uvparam_dict[uvparam_name] = "{}_ra{:.4f}_dec{:.4f}".format(
+            cat_name = "{}_ra{:.4f}_dec{:.4f}".format(
                 source_file_name, src.ra.deg[0], src.dec.deg[0]
             )
         else:
-            uvparam_dict[uvparam_name] = "unprojected"
+            cat_name = "unprojected"
     elif "object_name" in param_dict:
-        uvparam_dict[uvparam_name] = param_dict["object_name"]
+        cat_name = param_dict["object_name"]
     else:
-        uvparam_dict[uvparam_name] = param_dict["cat_name"]
+        cat_name = param_dict["cat_name"]
+    phase_center_catalog = {0: {"cat_name": cat_name, "cat_type": "unprojected"}}
 
     uv_obj = UVData()
```

Comment:
I think we can remove this line. We call UVData.new() in each block of this if statement.

Reply:
I use it to detect whether the UVData object has a "telescope" object attached to it (the hasattr call in the next line).

Reply:
oh yeah i do see that. Missed it before. Yeah I guess I'm not a fan of binding variables we're not using.